packages/core/test/insured-bridge/BridgePool.js

```diff
@@ -316,40 +316,58 @@ describe("BridgePool", () => {
       // Check event is logged correctly.
       await assertEventEmitted(txn, bridgePool, "DepositRelayed", (ev) => {
         return (
-          ev.sender === relayAncillaryData.l2Sender &&
-          ev.depositTimestamp === relayAncillaryData.depositTimestamp &&
-          ev.recipient === relayAncillaryData.recipient &&
-          ev.l1Token === relayAncillaryData.l1Token &&
+          ev.sender === relayData.l2Sender &&
+          ev.depositTimestamp === relayData.depositTimestamp &&
+          ev.recipient === relayData.recipient &&
+          ev.l1Token === relayData.l1Token &&
           ev.slowRelayer === relayer &&
-          ev.amount === relayAncillaryData.amount &&
-          ev.amount === relayAncillaryData.amount &&
-          ev.proposerRewardPct === relayAncillaryData.proposerRewardPct &&
-          ev.realizedFeePct === relayAncillaryData.realizedFeePct &&
+          ev.amount === relayData.amount &&
+          ev.amount === relayData.amount &&
+          ev.proposerRewardPct === relayData.proposerRewardPct &&
+          ev.realizedFeePct === relayData.realizedFeePct &&
           ev.depositContract === depositContractImpersonator
         );
       });
 
       // Check Deposit struct is stored correctly.
-      const deposit = await bridgePool.methods.deposits(relayAncillaryData.depositId).call({ from: relayer });
+      const relayAncillaryData = await bridgePool.methods.getRelayAncillaryData(relayData).call();
+      const deposit = await bridgePool.methods.deposits(relayAncillaryData).call();
       assert.equal(deposit.depositState, InsuredBridgeDepositStateEnum.PENDING_SLOW);
       assert.equal(deposit.depositType, InsuredBridgeDepositTypeEnum.SLOW);
       assert.equal(deposit.instantRelayer, ZERO_ADDRESS);
-      Object.keys(relayAncillaryData).forEach((key) => {
-        assert.equal(relayAncillaryData[key], deposit.relayData[key]);
-      });
+      const abiEncodedDataPacked = web3.eth.abi.encodeParameters(
+        ["uint64", "address", "address", "uint64", "address", "uint256", "uint64", "uint64", "uint64", "address"],
+        [
+          relayData.depositId,
+          relayData.l2Sender,
+          relayData.recipient,
+          relayData.depositTimestamp,
+          relayData.l1Token,
+          relayData.amount,
+          relayData.maxFeePct,
+          relayData.proposerRewardPct,
+          relayData.realizedFeePct,
+          relayData.slowRelayer,
+        ]
+      );
+      const expectedRelayHash = web3.utils.soliditySha3(abiEncodedDataPacked);
+      const relayHash = await bridgePool.methods.getRelayHash(relayData).call();
+      assert.equal(
```

Comment:
This test is worth looking at for how clients can use data hashed inside of the relay hash